pkg/api/api.go

```diff
@@ -126,6 +126,9 @@ func (hs *HTTPServer) registerRoutes() {
 	// api renew session based on cookie
 	r.Get("/api/login/ping", quota("session"), routing.Wrap(hs.LoginAPIPing))
 
+	// expose plugin file system assets
+	r.Get("/public/plugins/:pluginId/*", hs.GetPluginAssets)
```

Comment:
Thinking about it. This might require some other changes in regards to middleware, but not sure yet. 

This one perhaps shouldn't apply to plugin static assets:
https://github.com/grafana/grafana/blob/31a8413fd3fb8b9c9e321bcd4bd5072d0777d19c/pkg/middleware/middleware.go#L28-L47

Wonder if this one and the whole token rotation thingy would apply to this route?
https://github.com/grafana/grafana/blob/458681c4242022e802d1c1bac4358647778106b6/pkg/services/contexthandler/contexthandler.go#L289

Reply:
>As discussed, we might want to add the reqSignedIn middleware?

Ah I just made sure the change I made didn't alter the behaviour. But thinking about it, they do live under `public/*` so it would be strange to require auth for them in its current form. Not sure if there's a scenario where one might require access without being signed in, so will think about this further.

Great catch about the middlewares. Previously, it used none. Unless there's some security concerns, probably best to keep it that way - WDYT?

Reply:
I think keeping it public is fine -- but perhaps we should exclude executable files and maybe scripts?
https://play.grafana.org/public/plugins/grafana-googlesheets-datasource/gfx_sheets_linux_arm64

Reply:
Public should be fine